app/services/sync_service.rb

```diff
@@ -15,14 +15,18 @@ def sync(request_key, records, request_payload, report_errors_on_class: nil)
     begin
       request = to_request(request_key, records, request_payload)
       response = api_post("api/v1/#{request_key.to_s}/sync", Hash[request_key.to_sym, request])
+      unsynced_ids = []
       errors = JSON(response.body)['errors'].map do |error|
+        unsynced_ids << error['id']
         if report_errors_on_class.present?
           uuid_field = "#{request_key.to_s.singularize}_uuid"
           report_errors_on_class.find_by(Hash[uuid_field, error['id']]).attributes.merge(error: error.except('id'))
         else
           error
         end
       end
+
+      mark_as_synced(records, unsynced_ids)
```

Comment:
We can pass just the synced records to mark_as_synced. 
Passing unsynced_ids to marked as synced seems counter intuitive.

Reply:
Will update this - `unsynced_ids` will be used to mark records for which syncing failed.